packages/js-flagsmith-v1-provider/src/lib/flagsmith-v1.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  Context,
+  ContextTransformer,
+  FeatureProvider,
+  FlagEvaluationOptions,
+  FlagTypeError,
+  FlagValueParseError,
+  ProviderOptions,
+} from '@openfeature/openfeature-js';
+import * as flagsmith from 'flagsmith-nodejs';
+
+export interface FlagsmithProviderOptions extends ProviderOptions<string | undefined> {
+  environmentID: string;
+}
+
+/**
+ * Additional custom properties in Flagsmith are associated with the user (traits), and not passed directly to the flag evaluation.
+ * This is a very basic transformer function that defines Flagsmith traits based on the OpenFeature context. Note that it 
+ * doesn't handle nested properties.
+ * 
+ * We may want to generalize this concept, and provide a default for all providers.
+ */
+const DEFAULT_CONTEXT_TRANSFORMER = (context: Context): string | undefined => {
+  if (typeof context?.userId === 'string') {
+    Object.keys(context).forEach((key) => {
+      if (
+        key !== 'userId' &&
+        (typeof context[key] === 'boolean' ||
+          typeof context[key] === 'string' ||
+          typeof context[key] === 'number')
+      ) {
+        flagsmith.setTrait(context.userId as string, key, context[key]);
+      }
+    });
+  }
+  return context?.userId;
+};
```

Comment:
@beeme1mr this is a very simple "ContextTransformer". For Flagsmith, it reduce the context to a single string (the user id), after setting all other props defined in the context as traits. 

I guess this strategy isn't a problem. I assume since these methods aren't asynchronous, there's no chatter involved in setting traits, and it's only adding properties to a locally cached user until some server synchronization occurs, perhaps during flag evaluation. Maybe @dabeeeenster can confirm.

Of course I did test this, and the properties passed in the context are indeed added to the user and show up in the Flagsmith UI.

Reply:
That makes perfect sense!

I think I will add a v2 provider to this PR. It should be very similar to this one, but with the slight differences you mention. Until v2 is official, it can't hurt to have both, I think.

Reply:
I've added a flagsmith-v2 provider using the new beta SDK.